Google.Api.Generator/Generation/ServiceDetails.cs

```diff
@@ -62,6 +70,7 @@ public ServiceDetails(ProtoCatalog catalog, string ns, ServiceDescriptor desc, S
             SnippetsClientName = $"{desc.Name.ToLowerCamelCase()}Client";
             UnitTestsTyp = Typ.Manual(UnitTestsNamespace, $"Generated{desc.Name}ClientTest");
             NonStandardLro = NonStandardLroDetails.ForService(desc);
+            Mixins = serviceConfig?.Apis.Select(api => AvailableMixins.GetValueOrDefault(api.Name)).Where(mixin => mixin is object).ToList() ?? Enumerable.Empty<MixinDetails>();
```

Comment:
I'm not sure - a service yaml file can contain multiple services which aren't all mixins. Worth considering though - extraneous ones may be stripped anyway.

Reply:
If something is added in service yaml that is not a known mixin and we don't have it in the request to generate, it might be e.g. a misspelled mixin (`Locattions`).
Making a check here and failing might be a good idea.

Reply:
(This is now addressed in an additional commit.)